File: php-poo/aulas/aula06/ControleRemoto.php
<?php
require_once 'Controlador.php';
class ControleRemoto implements Controlador
{
    // Atributos
    private $volume;
    private $ligado;
    private $tocando;

    // Metodos Especiais
    public function __construct()
    {
        $this->volume = 0;
        $this->ligado = false;
        $this->tocando = false;
    }

    // Metodos Getters e Setters
    /**
     * @return int
     */
    private function getVolume(): int
    {
        return $this->volume;
    }

    /**
     * @param int $volume
     */
    private function setVolume(int $volume): void
    {
        $this->volume = $volume;
    }

    /**
     * @return false
     */
    private function getLigado(): bool
    {
        return $this->ligado;
    }

    /**
     * @param false $ligado
     */
    private function setLigado(bool $ligado): void
    {
        $this->ligado = $ligado;
    }

    /**
     * @return false
     */
    private function getTocando(): bool
    {
        return $this->tocando;
    }

    /**
     * @param false $tocando
     */
    private function setTocando(bool $tocando): void
    {
        $this->tocando = $tocando;
    }

    public function ligar()
    {
        // TODO: Implement ligar() method.
        $this->setLigado(true);
        $this->setVolume(50);
    }

    public function desligar()
    {
        // TODO: Implement desligar() method.
        $this->setLigado(false);
        $this->setTocando(false);
        $this->setVolume(0);
        // $this->setVolume(0);
    }

    public function abrirMenu()
    {
        // TODO: Implement abrirMenu() method.
        echo "<br> -------------------- MENU -------------------- </br>";

        echo "<br> Está ligado? " .($this->getLigado() ? "SIM" : "NÃO") ."<br>";

        echo "<br> Está tocando? " .($this->getTocando() ? "SIM" : "NÃO") ."<br>";

        echo "<br> Volume: " .$this->getVolume();
        for ($i = 0; $i <= $this->getVolume(); $i+=10 ) {
            echo "|";
        }
        echo "<br>";
    }

    public function fecharMenu()
    {
        // TODO: Implement fecharMenu() method.
        echo "<br> Fechando Menu... <br>";
    }

    public function maisVolume()
    {
        // TODO: Implement maisVolume() method.
        if ($this->getLigado()) {
            $this->setVolume($this->getVolume() + 10);
        } else {
             echo "<br> <span class='erro'> [ ERRO!] Não posso aumentar o volume. O controle não está ligado ! </span> <br> ";
        }
    }

    public function menosVolume()
    {
        // TODO: Implement menosVolume() method.
        if ($this->getLigado()) {
            $this->setVolume($this->getVolume() - 10);
        } else {
            echo "<br> <span class='erro'> [ ERRO!] Não posso diminuir o volume. O controle não está ligado ! </span> <br> ";
        }
    }

    public function ligarMudo()
    {
        // TODO: Implement ligarMudo() method.
        if ($this->getLigado() && $this->getVolume() > 0) {
            $this->setVolume(0);
        } else {
            echo "<br> <span class='erro'> [ ERRO!] Não posso ligar o mudo. O controle não está ligado ! </span> <br> ";
        }
    }

    public function desligarMudo()
    {
        // TODO: Implement desligarMudo() method.
        if ( ($this->getLigado()) && ($this->getVolume()) == 0 ) {
            $this->setVolume(50);
        }
    }

    public function play()
    {
        // TODO: Implement play() method.
        if ($this->getLigado() && !$this->getTocando()) {
            $this->setTocando(true);
        }
    }

    public function pause()
    {
        // TODO: Implement pause() method.
        if ($this->getLigado() && $this->getTocando()) {
            $this->setTocando(false);
        }
    }
}

File: java-poo/projects/poo-aula06/src/aula06/ControleRemoto.java
package aula06;

public class ControleRemoto implements Controlador {

    // Atributos

    private int volume;
    private boolean ligado;
    private boolean tocando;

    // Método Construtor

    public ControleRemoto() {
        this.volume = 10;
        this.ligado = false;
        this.tocando = false;
    }

    // Métodos Getters e Setters

    private int getVolume() {
        return volume;
    }

    private void setVolume(int volume) {
        this.volume = volume;
    }

    // type boolean ->  Generate (Default) -> isLigado()
    private boolean getLigado() {
        return ligado;
    }

    private void setLigado(boolean ligado) {
        this.ligado = ligado;
    }

    // type boolean ->  Generate (Default) -> isTocando()
    private boolean getTocando() {
        return tocando;
    }

    private void setTocando(boolean tocando) {
        this.tocando = tocando;
    }

    /**
     * @Override = Sobrescrever; Sobrepor; Sobrescrita
     *
     * @Override -> Significa que eu já tinha um método abstrato (abstract) e sobrescrevo os métodos.
     *
     * A sobrescrita (ou override) está diretamente relacionada à orientação a objetos, mais especificamente com a herança.
     * Com a sobrescrita, conseguimos especializar os métodos herdados das superclasses, alterando o seu comportamento nas subclasses por um mais específico.
     */

    // Métodos Abstratos

    @Override
    public void ligar() {
        this.setLigado(true);
    }

    @Override
    public void desligar() {
        this.setLigado(false);
    }

    @Override
    public void abrirMenu() {
        if ( this.getLigado() ) {
            System.out.println("\nAbrindo Menu!");
            System.out.println("----------------------------------------------");
            System.out.println("Está ligado? " + this.getLigado());
            System.out.println("Está tocando? " + this.getTocando());
            System.out.print("Volume: " + this.getVolume() + " -> ");
            for (int i = 0; i < this.getVolume(); i += 10) {
                System.out.print(" ▲ ");
            }
            System.out.println("\n----------------------------------------------");
        } else {
            System.out.println("Não é possível mostrar o Menu, pois o controle remoto está desligado!");
        }
    }

    @Override
    public void fecharMenu() {
        System.out.println("Fechando Menu!");
    }

    @Override
    public void maisVolume() {
        if ( this.getLigado() ) {
            this.setVolume(this.getVolume() + 10);
        } else {
            System.out.println("Não é possível aumentar o volume!");
        }
    }

    @Override
    public void menosVolume() {
        if ( this.getLigado() ) {
            this.setVolume(this.getVolume() - 10);
        } else {
            System.out.println("Não é possível diminuir o volume!");
        }
    }

    @Override
    public void ligarMudo() {
        if ( this.getLigado() && this.getVolume() > 0 ) {
            this.setVolume(0);
        } else {
            System.out.println("Não é possível ligar o mudo!");
        }
    }

    @Override
    public void desligarMudo() {
        if ( this.getLigado() && this.getVolume() == 0 ) {
            this.setVolume(50);
        }
    }

    @Override
    public void play() {
        if ( this.getLigado() && !(this.getTocando()) ) {
            this.setTocando(true);
        } else {
            System.out.println("Não é possível executar o play!");
        }
    }

    @Override
    public void pause() {
        if ( this.getLigado() && this.getTocando() ) {
            this.setTocando(false);
        } else {
            System.out.println("Não é possível executar o pause!");
        }
    }
}
